Pod/Classes/LSClockState.m

```diff
@@ -90,10 +90,11 @@ - (void)addSampleWithOriginMicros:(SInt64)originMicros
     // Note that self.samplesQueue is DISPATCH_QUEUE_SERIAL and it makes thread safe reads/writes.
     dispatch_async(self.samplesQueue, ^{
         // Discard the oldest sample and push the new one.
-        [self.samples removeObjectAtIndex:0];
-        [self.samples
-         addObject:[[LSSyncSample alloc] initWithDelayMicros:latestDelayMicros offsetMicros:latestOffsetMicros]];
-
+        if (self.samples.count > 0) {
```

Comment:
yeah, I'm just thinking out loud. this is not an issue with your changes. I looked after I wrote the comment and have some more context on the motivation here. TL;DR it's reasonable, though we're not taking advantage of the array at the moment... basically just future-proofing.